doc_guide/src/generating_markdown.md

```diff
@@ -0,0 +1,50 @@
+# Generating Markdown
+
+Depending on your case you can interact with `TypeDescription`s in multiple ways.
+
+If you are given a JSON formatted type description file, then you should use the CLI for further processing.
+(This may be the case if you are using someone else's project).
+
+If you have some Rust types and want to generate markdown directly from their
+`TypeDescription` then you might be best served with the library.
+
+## Using the CLI
+
+The preferred way of interacting with the CLI is to use the `type_description` binary.
+
+It can be invoked in the following ways:
+
+- With the nix package manager (flakes enabled): `nix run github:TheNeikos/type_description`
+- With cargo: `cargo install type_description`
+- Directly from source: `git clone https://github.com/TheNeikos/type_description && cd type_description && cargo run -F bin`
+
+Either way you have installed it. You can now generate markdown from it by sending it to standard input.
+
+E.g.:
+
+- Using nix: `cat desc.rtd | nix run github:TheNeikos/type_description`
```

Comment:
What is `desc.rtd`?

Reply:
Haha, just a made up filename of where the 'rust type description' could be. I'll change it!

Reply:
Yeah, it is somewhat surprising. I actually looked at the repository whether there's a file named `desc.rtd`. I also am not sure whether the `example.com` is a good idea... maybe replace it by something like `<your description file>` or `<your description URL>`!